hack/test-olm.sh

```diff
@@ -0,0 +1,285 @@
+#!/bin/bash
+
+set -ex
+
+OPT=${1:-kind}
+
+if [[ $OPT != "kind" ]] && [[ $OPT != "minikube" ]]; then
+	echo "error: argument must be either kind or minikube"
+	exit 1
+fi
+
+LOCAL_OLM=olm-repo
+LOCAL_OPREG=operator-registry
+
+case "$OPT" in
+	minikube)
+		LOCAL_REGISTRY=localhost.localdomain:5000
+		HOST_IP=$(hostname -I | awk '{ print $1 }')
+		;;
+
+	kind)
+		LOCAL_REGISTRY=localhost:5000
+		;;
+esac
+
+NMOBUNDLE=node-maintenance-operator-bundle
+NMOREG=node-maintenance-operator-registry
+NMO=node-maintenance-operator
+CATALOG_NAMESPACE="olm"
+REGISTRY=quay.io/kubevirt
+
+log_to_file() {
+	exec &> >(tee -a test-olm.out)
+}
+
+start_registry() {
+	hreg_name='local-registry'
+	reg_port='5000'
+	running="$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true)"
+	if [ "${running}" != 'true' ]; then
+	  docker run \
+		-d --restart=always -p "${reg_port}:5000" --name "${reg_name}" \
+		registry:2
+	fi
+}
+
+setup_src() {
+	if [[ ! -f $LOCAL_OLM/repo_init ]]; then
+		git clone https://github.com/operator-framework/operator-lifecycle-manager.git $LOCAL_OLM
```

Comment:
What do you need this for? You can install OLM with e.g operator sdk.

Reply:
need to run make run-local to get the olm test environment for olm on openshift 4.5 (at least that's what Alex Greene told me)

Reply:
make run-local builds and deploys from sources. We don't need that. We just want to deploy a released version.

Reply:
The release downloaded by the latest script still has a problem: the client configuration section of the wehookconfiguration object that has been generated by olm refers to the wrong port number. Have asked a question on the olm channel and am waiting for a response.

Reply:
Alex Greene (from OLM team) says he knows the latest official release doesn't include a bugfix that would sets the port number correctly. that one will be included with the next release. So until that release comes out the best we can do is to clone the OLM repository for the test environment.